libraries/AP_HAL_SITL/sitl_barometer.cpp

```diff
@@ -76,7 +76,9 @@ void SITL_State::_update_barometer(float altitude)
 
     // find data corresponding to delayed time in buffer
     for (uint8_t i=0; i<=baro_buffer_length-1; i++) {
-        time_delta_baro = abs(delayed_time_baro - buffer_baro[i].time); // find difference between delayed time and time stamp in buffer
+        // find difference between delayed time and time stamp in buffer
+        time_delta_baro = abs(
+                (int32_t)(delayed_time_baro - buffer_baro[i].time));
```

Comment:
casting to int32_t seems wrong in here. The result of the subtraction of uint32 - uint32 can still be a number that does not fit into int32. In that case, the resulting positive number would be cast into a negative number, yielding the wrong result.